### PR TITLE
[tests-only] [full-ci] Added `Then` steps to improve checks in tests for apiComments suite

### PR DESCRIPTION
**Pull request by grgprarup**
## Description
This PR adds `Then` steps to existing API test scenarios to better check the results of `When` steps.

**Suites covered:**
- `apiCapabilities`
- `apiComments`

## Related Issue

- Part of https://github.com/owncloud/core/issues/39512

## How Has This Been Tested?

- Locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Review by kiranparajuli589 — COMMENTED** [review]
lgtm :+1: please squash your commits before merge.

**Review by phil-davis — DISMISSED** [review]
These are going to be annoying for expected-failures updates. So we will need to be good at updating expected-failures very regularly to minimize the pain.

File: tests/acceptance/features/apiComments/deleteComments.feature [hunk 1]
```diff
@@ -98,4 +99,5 @@ Feature: Comments
     And user "Alice" has been deleted
     And user "Alice" has been created with default attributes and without skeleton files
     When user "Alice" creates folder "/FOLDER_TO_COMMENT" using the WebDAV API
-    Then user "Alice" should have 0 comments on folder "/FOLDER_TO_COMMENT"
+    And the HTTP status code should be "201"
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiComments/deleteComments.feature:102]:
```suggestion
    Then the HTTP status code should be "201"
```

**Review by kiranparajuli589 — APPROVED** [review]

**Comment by kiranparajuli589** [comment]
failed at `/drone/src/tests/acceptance/features/webUIUpload/upload.feature:131`
https://drone.owncloud.com/owncloud/core/34667/151/17

build restarted

**Pull request merged**